### PR TITLE
Adding Github Actions to build and Publish

### PR DESCRIPTION
**Pull request by akhilsunilkumar**

### CODE REVIEW

File: .github/workflows/gradle-publish.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    - uses: actions/checkout@v2
+    - name: Set up JDK 11
+      uses: actions/setup-java@v2
+      with:
+        java-version: '11'
+        distribution: 'temurin'
+        server-id: github # Value of the distributionManagement/repository/id field of the pom.xml
+        settings-path: ${{ github.workspace }} # location for the settings.xml file
+        
+    - name: Build Skipping Tests with Gradle
```
Comment (ankur4u007) [inline .github/workflows/gradle-publish.yml]: Please change the name to `Assemble binaries on tags`

File: .github/workflows/gradle-publish.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      if: github.ref == 'refs/heads/main'
+      uses: gradle/gradle-build-action@937999e9cc2425eddc7fd62d1053baf041147db7
+      with:
+        arguments: clean assemble
+        
+    - name: Build with Gradle
```
Comment (ankur4u007) [inline .github/workflows/gradle-publish.yml]: Please change the name to `Build on Pull requests or merges`

File: .github/workflows/gradle-publish.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      if: ${{ github.event_name == 'pull_request' || github.event.pull_request.merged == 'true' }}
+      uses: gradle/gradle-build-action@937999e9cc2425eddc7fd62d1053baf041147db7
+      with:
+        arguments: clean build
+      
+    - name: Decode and save GPG Secret File from secrets
```
Comment (ankur4u007) [inline .github/workflows/gradle-publish.yml]: Please merge this and ` publish to mavencentral` in single step.

File: app/build.gradle.kts [hunk 4]
```diff
@@ -13,3 +13,14 @@ dependencies {
     implementation("org.springframework.data:spring-data-elasticsearch:_")
     implementation("org.springframework:spring-web:_")
 }
+
+project.configurations {
```
Comment (ankur4u007) [inline app/build.gradle.kts]: Can you clarify why is this change required ?

File: build.gradle.kts [hunk 5]
```diff
@@ -13,11 +11,10 @@ plugins {
     idea
     jacoco
     `maven-publish`
+    signing
     kotlin("jvm")
     id("org.jlleitschuh.gradle.ktlint")
     id("io.gitlab.arturbosch.detekt")
-    id("com.jfrog.artifactory")
```
Comment (ankur4u007) [inline build.gradle.kts]: 👍

File: build.gradle.kts [hunk 6]
```diff
@@ -104,7 +100,8 @@ subprojects {
     // detekt
     detekt {
         buildUponDefaultConfig = true // preconfigure defaults
-        config = files("$rootDir/gradle/detekt.yml") // point to your custom config defining rules to run, overwriting default behavior
+        config =
+            files("$rootDir/gradle/detekt.yml") // point to your custom config defining rules to run, overwriting default behavior
```
Comment (ankur4u007) [inline build.gradle.kts]: I think it should be on same line ?

File: build.gradle.kts [hunk 7]
```diff
@@ -135,49 +132,115 @@ subprojects {
...
+                        url.set("https://github.com/lowes/auditor")
+                        licenses {
+                            license {
+                                name.set("Apache-2.0")
+                                url.set("https://opensource.org/licenses/Apache-2.0")
+                            }
+                        }
+                        developers {
+                            developer {
+                                id.set("akhilsunilkumar")
```
Comment (ankur4u007) [inline build.gradle.kts]: we should limit the names to the lowe's open source contact and avoid any personal names here.

**Review by ankur4u007 — CHANGES_REQUESTED** [review]
@akhilsunilkumar  I have few comments, PTAL. LGTM overall.

**Comment by ankur4u007** [comment]
@akhilsunilkumar Also please clean up your commits, may be squash them in a single commit with proper messaging.

File: app/build.gradle.kts (hunk 4 above)
Comment (akhilsunilkumar) [reply app/build.gradle.kts]: this was added because the fat jar was not getting created for the App module

File: build.gradle.kts (hunk 6 above)
Comment (akhilsunilkumar) [reply build.gradle.kts]: I'll Add it back

File: build.gradle.kts (hunk 7 above)
Comment (akhilsunilkumar) [reply build.gradle.kts]: Ok.. I'll Add Just Lowes

File: .github/workflows/gradle-publish.yml (hunk 1 above)
Comment (akhilsunilkumar) [reply .github/workflows/gradle-publish.yml]: ok

File: .github/workflows/gradle-publish.yml (hunk 2 above)
Comment (akhilsunilkumar) [reply .github/workflows/gradle-publish.yml]: ok

File: .github/workflows/gradle-publish.yml (hunk 3 above)
Comment (akhilsunilkumar) [reply .github/workflows/gradle-publish.yml]: ok
